guarddog/analyzer/metadata/pypi/repository_integrity_mismatch.py

```diff
@@ -0,0 +1,260 @@
+            if file_name not in files:  # ignore files we don't have in the distribution
+                continue
+            repo_hash, repo_content = get_file_hash(os.path.join(repo_root, file_name))
+            pkg_hash, pkg_content = get_file_hash(os.path.join(root, file_name))
+            if repo_hash != pkg_hash:
+                if exclude_result(file_name, repo_root, root):
+                    continue
+                res = {
+                    "file": os.path.join(relative_path, file_name),
+                    "repo_sha256": repo_hash,
+                    "pkg_sha256": pkg_hash
+                }
+                mismatch.append(res)
+    return mismatch
+
+
+def find_suitable_tags_in_list(tags, version):
+    tag_candidates = []
+    for tag_name in tags:
+        if tag_name.endswith(version):
+            tag_candidates.append(tag_name)
+    return tag_candidates
+
+
+def find_suitable_tags(repo, version):
+    tags_regex = re.compile('^refs/tags/(.*)')
+    tags = []
+    for ref in repo.references:
+        match = tags_regex.match(ref)
+        if match is not None:
+            tags.append(match.group(0))
+
+    return find_suitable_tags_in_list(tags, version)
+
+
+# Note: we should have the GitHub related logic factored out as we will need it when we check for signed commits
+class PypiIntegrityMismatch(IntegrityMismatch):
+    """This package contains files that have been tampered with between the source repository and the package CDN"""
+    RULE_NAME = "repository_integrity_mismatch"
+
+    def detect(self, package_info, path: Optional[str] = None, name: Optional[str] = None,
+               version: Optional[str] = None) -> tuple[bool, str]:
+        if name is None:
+            raise Exception("Detector needs the name of the package")
+        if path is None:
+            raise Exception("Detector needs the path of the package")
+        # let's extract a source repository (GitHub only for now) if we can
+        github_urls, best_github_candidate = find_github_candidates(package_info)
+        if len(github_urls) == 0:
+            return False, "Could not find any GitHub url in the project's description"
+        # now, let's find the right url
+
+        github_url = find_best_github_candidate((github_urls, best_github_candidate), name)
+
+        if github_url is None:
+            return False, "Could not find a good GitHub url in the project's description"
+
+        # ok, now let's try to find the version! (I need to know which version we are scanning)
+        if version is None:
+            version = package_info["info"]["version"]
+        if version is None:
+            raise Exception("Could not find suitable version to scan")
+        tmp_dir = os.path.dirname(path)
+        if tmp_dir is None:
+            raise Exception("no current scanning directory")
+
+        repo_path = os.path.join(tmp_dir, "sources", name)
+        repo = pygit2.clone_repository(url=github_url, path=repo_path)
```

Comment:
Testing on `datadog-lambda-python` package, this throws the following Exception:

```
'authentication required but no callback set'
``` 

... which is also never surfaced in the CLI/JSON output. Same for other packages e.g. `requests`

Reply:
OK so this was caused by my git config:

```
[url "git@github.com:"]
    insteadOf = https://github.com/
```

Probably worth investigating how to not honor the local .gitconfig file